src/Services/Formula/FormulaGrammar.vala

```diff
@@ -1,16 +1,34 @@
 using Spreadsheet.Services.Parsing;
 
 public class Spreadsheet.Services.Formula.FormulaGrammar : Grammar {
+    private string func_name_regex = "";
+
     public FormulaGrammar () {
         rules["root"] = root_rules ();
     }
 
+    private string get_func_name_regex () {
+        if (func_name_regex != "") {
+            return func_name_regex;
+        }
+ 
+        for (int i = 0; i < App.functions.size; i++) {
+            func_name_regex += "%s|%s".printf (App.functions[i].name, App.functions[i].name.ascii_up ());
+
+            if (i + 1 != App.functions.size) {
+                func_name_regex += "|";
+            }
+        }
```

Comment:
Couldn't this function be used instead of this loop? https://valadoc.org/glib-2.0/string.joinv.html